### ISSUE TITLE
Temporarily removing the time-range parameters to the Evict op

### ISSUE THREAD

**Issue by jarohen** [xtdb/xtdb#428]
We're going to temporarily remove the time-range parameters to the Evict operation, because the current semantics are inconsistent and buggy - this requires more thought before we can commit to it as a public API.

To be removed:
* valid-time/end-valid-time
* keep-earliest/keep-latest

To be kept:
* Evict all versions of a document, for all valid times

Related cards, out of scope of this one:
* #371 
* #375

Things to consider:
* How will this affect existing evict ops in live Kafka logs?

**Comment by jarohen** [comment]
@hraberg @refset let's confirm the backward-compatibility behaviour we require for existing evict ops in live Kafka logs

**Comment by hraberg** [comment]
This card doesn't currently deal with the race condition which would happen if someone tries to resurrect a document in quick succession of evicting it. We can split that work to a different card if necessary, as it still requires some analysis and discussion.

My basic idea is this:
1. Record evicted entity ids, these would simply be key-values stored by the evict op in a new internal index.
2. If a document arrives for an entity which has been marked as evicted, drop it instead of indexing and storing it, and emit a new tombstone to get rid of it again on Kafka. Log at debug or warn.
3. On the transaction side, potentially blow up the transaction in a general pre-condition if an operation involves an evicted event, alternatively log a warning and do nothing.
4. One could potentially introduce a new operation `:crux.tx/unevict` to allow writes to the entity id again at a later stage, but think that would be a separate card and discussion for sure. Can easily imagine opening up for various new race conditions.

**Comment by hraberg** [comment]
Note, the reason we cannot simply allow a new put to resurrect it is that there's a race condition between the topics, as an eviction might have overwritten its document before it gets indexed, or vice versa for the tombstone.

The above solution actually doesn't care about if the events happen close in time, as it's designed to converge to the same result regardless.

**Comment by jarohen** [comment]
> This card doesn't currently deal with the race condition which would happen if someone tries to resurrect a document in quick succession of evicting it. We can split that work to a different card if necessary

agree that we should split that out into a separate card and link it in the 'related' section here, if there's not already a card for it?

**Comment by hraberg** [comment]
Pretty sure there's no other card, it was discussed on Slack a bit in the context of this card.

**Comment by jarohen** [comment]
proposal re evicting/kafka log:
* if we spot an eviction op with time ranges on the Kafka log, we throw an exception and kill the indexer. the error message informs the user that they need to specify an env var flag to continue
* user supplies an env-var flag, something like `CRUX_EVICT_TIME_RANGES=EVICT_ALL`, declaring that they're happy with Crux evicting the whole entity
* if the env var is present, the Crux indexer continues indexing with the evict-all behaviour

cc @refset

**Comment by hraberg** [comment]
This has been mentioned elsewhere, like on #371 but this should preferably also use a lower-level API than the history API, see `crux.index/all-keys-in-prefix` and scan out the existing content hashes from the AEV index for `:crux.db/id` and evict them. Can be solved here or in the other card.

**Comment by jarohen** [comment]
agreed - this is in scope for #371 rather than here
